### PR TITLE
docs: Clearer description on when to use online backups

### PR DESCRIPTION
**Pull request by laysakura**
## Problem

`backup-restore.md` could be read as the following:

- It is safe to make backups for MySQL and PostgreSQL (both are under a Scalar DB instance) using `mysqldump` and `pg_dump` while the Scalar DB instance is accepting new transactions

But it should (quite naturally) cause inconsistent tables among the MySQL and PostgreSQL.

## What I did

- Divide two ways to create backups: online or using pause duration
- Add a flowchart for when to use online backup
- Clearly write the responsibility of creating an online backup belongs to users
- Break "JDBC database" sections into ones for each DB product

### CODE REVIEW

**Review by Torch3333 — APPROVED** [review]
LGTM, thank you!

**Comment by brfrn169** [comment]
@laysakura Sorry, I didn't tell you it, but can you please merge your PR after all the reviewers approves it?

**Comment by laysakura** [comment]
@brfrn169 Oh, sorry. I got it.

**Comment by brfrn169** [comment]
@laysakura No problem 🙇

**Review by brfrn169 — COMMENTED** [review]
LGTM! Thank you!

**Review by brfrn169 — APPROVED** [review]

**Review by komamitsu — APPROVED** [review]
LGTM, thanks!

File: docs/backup-restore.md [hunk 1]
```diff
@@ -5,19 +5,41 @@ This document sets out some guidelines for backing up and restoring the database
...
 
-You can take a backup with your favorite way for JDBC databases.
-One requirement for backup in Scalar DB on JDBC databases is that backups for all the Scalar DB managed tables (including the coordinator tables) need to be transactionally-consistent or automatically recoverable to a transactionally-consistent state.
+### Flowchart
+
+- :question: Use a single DB under Scalar DB?
+  - :question: The DB has transaction support?
+    - :arrow_right: You may use [online backup](#online-backup)
+- Otherwise
+  - :arrow_right: You may use [pause duration](#transactionally-consistent-backups-during-pause-duration)
```
Comment (feeblefakie) [inline docs/backup-restore.md:18]:
```suggestion
  - :arrow_right: use [backup with explicit pausing](#backup-with-explicit-pausing)
```

File: docs/backup-restore.md [hunk 2]
```diff
@@ -5,19 +5,41 @@ This document sets out some guidelines for backing up and restoring the database
...
+
+### Online Backup
```
Comment (feeblefakie) [inline docs/backup-restore.md:20]:
```suggestion
### Backup without explicit pausing
```

File: docs/backup-restore.md [hunk 3]
```diff
@@ -5,19 +5,41 @@ This document sets out some guidelines for backing up and restoring the database
...
+#### PostgreSQL (backup)
+
+Use the `pg_dump` command.
 
-#### Basic strategy to create a transactionally-consistent backup
+#### Amazon RDS or Azure Database for MySQL/PostgreSQL (backup)
+
+You can restore to any point within the backup retention period with the automated backup feature.
+
+### Transactionally-consistent backups during pause duration
```
Comment (feeblefakie) [inline docs/backup-restore.md:42]:
```suggestion
### Backup with explicit pausing
```

File: docs/backup-restore.md [hunk 4]
```diff
@@ -47,35 +73,40 @@ Please see [the doc](https://github.com/scalar-labs/cassy/blob/master/docs/getti
...
 You must enable the point-in-time recovery (PITR) feature for DynamoDB tables. If you use [Scalar DB Schema Loader](https://github.com/scalar-labs/scalardb/tree/master/schema-loader) for creating schema, it enables PITR feature for tables by default.
-To specify a transactionally-consistent restore point, please pause the Scalar DB application of a DynamoDB as described in [the basic strategy](#basic-strategy-to-create-a-transactionally-consistent-backup).
+To specify a transactionally-consistent restore point, please pause the Scalar DB application of a DynamoDB as described [here](#transactionally-consistent-backups-during-pause-duration).
 
 ## Restore Backup
 
-### JDBC databases
+### MySQL (restore)
+
+If you use MySQL and `mysqldump` to create a backup file, use `mysql` command to restore the file as specified [in the MySQL doc](https://dev.mysql.com/doc/mysql-backup-excerpt/8.0/en/reloading-sql-format-dumps.html).
```
Comment (feeblefakie) [inline docs/backup-restore.md:87]:
```suggestion
If you use `mysqldump` to create a backup file, use `mysql` command to restore the file as specified [in the MySQL doc](https://dev.mysql.com/doc/mysql-backup-excerpt/8.0/en/reloading-sql-format-dumps.html).
```

File: docs/backup-restore.md [hunk 5]
```diff
@@ -47,35 +73,40 @@ Please see [the doc](https://github.com/scalar-labs/cassy/blob/master/docs/getti
...
+
+### PostgreSQL (restore)
+
+If you use PostgreSQL and `pg_dump` to create a backup file, use `psql` command to restore the file as specified in [the PostgreSQL doc](https://www.postgresql.org/docs/current/backup-dump.html#BACKUP-DUMP-RESTORE).
```
Comment (feeblefakie) [inline docs/backup-restore.md:91]:
```suggestion
If you use `pg_dump` to create a backup file, use `psql` command to restore the file as specified in [the PostgreSQL doc](https://www.postgresql.org/docs/current/backup-dump.html#BACKUP-DUMP-RESTORE).
```

File: docs/backup-restore.md [hunk 6]
```diff
@@ -5,19 +5,41 @@ This document sets out some guidelines for backing up and restoring the database
...
+In some cases, you may create a transactional backup of your databases online.
 
-#### JDBC databases
+In other cases, it is recommended to first make Scalar DB drain outstanding requests and stop accepting new requests and then make the backups (see [here](#transactionally-consistent-backups-during-pause-duration) for detail).
 
-You can take a backup with your favorite way for JDBC databases.
-One requirement for backup in Scalar DB on JDBC databases is that backups for all the Scalar DB managed tables (including the coordinator tables) need to be transactionally-consistent or automatically recoverable to a transactionally-consistent state.
+### Flowchart
+
+- :question: Use a single DB under Scalar DB?
```
Comment (feeblefakie) [inline docs/backup-restore.md:14]:
```suggestion
- :question: Do you use a single DB under Scalar DB?
```

File: docs/backup-restore.md [hunk 7]
```diff
@@ -5,19 +5,41 @@ This document sets out some guidelines for backing up and restoring the database
...
+  - :question: The DB has transaction support?
```
Comment (feeblefakie) [inline docs/backup-restore.md:15]:
```suggestion
  - :question: Does the DB have transaction support?
```

File: docs/backup-restore.md [hunk 8]
```diff
@@ -5,19 +5,41 @@ This document sets out some guidelines for backing up and restoring the database
...
+If you use a single DB under Scalar DB and it has transaction support, it could be possible to create backups even while Scalar DB continues to accept transactions.
+
+One requirement for backup in Scalar DB is that backups for all the Scalar DB managed tables (including the coordinator tables) need to be transactionally-consistent or automatically recoverable to a transactionally-consistent state.
 That means that you need to create a consistent snapshot by dumping all tables in a single transaction.
-For example, you can use the `mysqldump` command with `--single-transaction` option in MySQL and the `pg_dump` command in PostgreSQL to achieve that.
-Or when you use Amazon RDS (Relational Database Service) or Azure Database for MySQL/PostgreSQL, you can restore to any point within the backup retention period with the automated backup feature, which satisfies the requirement.
 
-### For Non-transactional Databases
+The ways to create transactional backups are different among each database product.
+We show some examples for some database products but you are requested to find the safe ways to a transactional backup from your databases at your own risk.
```
Comment (feeblefakie) [inline docs/backup-restore.md:28]:
```suggestion
We show examples for some database products, but you are requested to find safe ways for a transactional backup from your databases at your own risk.
```

File: docs/backup-restore.md [hunk 9]
```diff
@@ -30,15 +52,19 @@ Note that when you use a point-in-time-restore/recovery mechanism, it is recomme
 Otherwise, the time you get as a paused duration might be too different from the time in which the pause was actually conducted, which could restore to a point where ongoing transactions exist.
 Also, it is recommended to pause a long enough time (e.g., 10 seconds) and use the mid-time of the paused duration as a restore point since clock synchronization cannot perfectly synchronize clocks between nodes.
 
-#### Database-specific ways to create a transactionally-consistent backup   
+#### Database-specific ways to create a transactionally-consistent backup
+
+**Databases with transaction support**
```
Comment (feeblefakie) [inline docs/backup-restore.md:57]: This and the following line may seem unnecessary.

**Review by feeblefakie — COMMENTED** [review]
Thank you very much for the heads-up.
Although the updates make things way clearer, I left several comments and suggestions to improve further.
PTAL when you get a chance!

File: docs/backup-restore.md (hunk 9 above)
Comment (laysakura) [reply docs/backup-restore.md]: 70bf1ec019756d461ea588bd060a3b5ed15293c1

File: docs/backup-restore.md [hunk 10]
```diff
@@ -5,19 +5,41 @@ This document sets out some guidelines for backing up and restoring the database
...
-#### JDBC databases
+In other cases, it is recommended to first make Scalar DB drain outstanding requests and stop accepting new requests and then make the backups (see [here](#transactionally-consistent-backups-during-pause-duration) for detail).
 
-You can take a backup with your favorite way for JDBC databases.
-One requirement for backup in Scalar DB on JDBC databases is that backups for all the Scalar DB managed tables (including the coordinator tables) need to be transactionally-consistent or automatically recoverable to a transactionally-consistent state.
+### Flowchart
+
+- :question: Use a single DB under Scalar DB?
+  - :question: The DB has transaction support?
+    - :arrow_right: You may use [online backup](#online-backup)
```
[the remark this answers is not in the document]
Comment (brfrn169) [reply docs/backup-restore.md:15]: @feeblefakie @laysakura Sounds good to me. Thank you!

**Comment by laysakura** [comment]
@feeblefakie PTAL again!

**Review by feeblefakie — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**